python/nicenicks/nicenicks.py

```diff
@@ -220,69 +248,55 @@ def nicedebug_command(word, word_eol, userdata):
-    #chanmsg: [':epitaph!~epitaph@CPE00a0241892b7-CM014480119187.cpe.net.cable.rogers.com', 'PRIVMSG', '#werd', ':mah', 'script', 'is', 'doing', 'stuff.']
-    #action:  [':rlz!railz@bzq-199-176.red.bezeqint.net', 'PRIVMSG', '#werd', ':\x01ACTION', 'hugs', 'elhaym', '\x01']
-    #private: [':olene!oqd@girli.sh', 'PRIVMSG', 'epinoodle', ':hey']
-
     if nicenicks_enabled:
-
-        dmsg(word, "RAWCOMMAND")
-        dmsg("COLORTABLE length = %d" % len(chancolortable), "RAWCOMMAND")
-
-        if word[2][0] != "#":
-            dmsg("Private message -- handing event back to X-Chat", "MSG_EVENT")
-            return EAT_NONE
-
-        chan = word[2]
-
-        if not chancolortable.has_key(chan):
+        dmsg("COLORTABLE length = %d" % len(chancolortable), "PRINTEVENT")
+        event_name = userdata
+        nick = word[0]
+        nick = hexchat.strip(nick, -1, 1) # remove existing colours
+
+        # This bit prevents infinite loops.
+        # Assumes nicks will never normally begin with "\017".
+        if nick.startswith(ec["o"]):
+            # We already did this event and are seeing it again, because this function gets triggered by events that even it generates.
+            dmsg("Already-processed nick found: " + repr(nick), "LOOP")
+            return hexchat.EAT_NONE
+
+
+        chan = hexchat.get_info("channel")
+        net = hexchat.get_info("network")
+        if not chancolortable.has_key((net, chan)):
             # make new color table
-            dmsg("Making new color table for "+chan, "COLORTABLE");
-            chancolortable[chan] = defaultcolortable[:]
+            dmsg("Making new color table for "+chan, "COLORTABLE")
+            chancolortable[net, chan] = defaultcolortable[:]
         else:
-            dmsg("Found COLORTABLE of length "+str(len(chancolortable[chan]))+" for channel "+chan, "COLORTABLE")
-
-        # get COLORTABLE for this channel
-        ctable = chancolortable[chan]
-        dmsg("COLORTABLE for "+chan+" = " + str(ctable), "COLORTABLE")
-
-        userinfo = userparser.match(word[0]).groups()
-        if len(userinfo) == 3:
-            nick = userinfo[0]
-            color = get_color(ctable, nick)
-            c = str(color)
-
-            if word[3] == ':\x01ACTION':
-                action = " ".join(word[4:])
-                fprint("%C"+c+"*$t"+nick+" "+action)
-            else:
-                fprint("%C2<%O%C"+c+nick+"%C2>%O$t"+word_eol[3][1:]+"%O")
-
-        return EAT_XCHAT # tell xchat to eat itself
-
+            dmsg("Found COLORTABLE of length "+str(len(chancolortable[net, chan]))+" for channel "+chan+" on network "+net, "COLORTABLE")
+        ctable = chancolortable[net, chan]
+        dmsg("COLORTABLE for "+chan+" on "+net+" = " + str(ctable), "COLORTABLE")
+        color = get_color(ctable, nick)
+        newnick = ecs('o') + col(color) + nick
+        word[0] = newnick
+        hexchat.emit_print(event_name, *word)
+        return hexchat.EAT_HEXCHAT
```

Comment:
After looking at the docs for the Python interface, I'm not sure whether this should be `EAT_ALL` instead.